LanguageFeatures/Extension-types/static_analysis_member_invocation_A01_t03.dart

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Consider an invocation of the extension type member m on the
+/// receiver expression e according to the extension type declaration V with the
+/// actual type arguments T1, ..., Ts. If the invocation includes an actual
+/// argument part (possibly including some actual type arguments) then call it
+/// args. If the invocation does not include an actual argument part, but it
+/// does include a list of actual type arguments, call it typeArgs. Finally,
+/// assume that V declares the type variables X1, ..., Xs.
+///
+/// Note that if the name of m is a name in the interface of Object (that is,
+/// toString, ==, hashCode, runtimeType, or noSuchMethod), the denoted member is
+/// necessarily a non-extension type member, which determines the static
+/// analysis and dynamic semantics.
+///
+/// @description Checks that members of an `Object` are treated as all other
+/// non-extension type members.
+/// @author sgrekhov22@gmail.com
+/// @issue 53740
+
+// SharedOptions=--enable-experiment=inline-class
+
+import "../../Utils/expect.dart";
+
+String log = "";
+
+class A {
+}
+
+class B implements A {
+  @override
+  int get hashCode {
+    log += "B.hashCode";
+    return super.hashCode;
+  }
+
+  @override
+  Type get runtimeType {
+    log += "B.runtimeType";
+    return super.runtimeType;
+  }
+
+  @override
+  bool operator ==(Object? other) {
```

Comment:
We could use a covariant parameter here, in order to verify that the static analysis takes this member signature into account at a call site (just like the `e1`/`e2` examples in other tests), and it doesn't just rely on the signature in `Object`.

Reply:
The variant that has a covariant parameter type would probably need to be in a different library, because we'd (most likely) need to test that statically using some situations that cause a compile-time error (like `myET1 == 42` where the parameter type of `myET1` is `ET1`).

Reply:
Thanks!